Adding MenuBar #68

kikilondon opened the pull request:
#68

pablisch commented on frontend/src/components/MenuBar/MenuBar.js:
```diff
@@ -0,0 +1,31 @@
+    justifyContent: "center",
+    alignItems: "center",
+    padding: 20,
+    backgroundColor: "#A9A9A9" // I cannot set this color as globalStyleVariable.js background
```
To do this, choose an appropriates name, e.g. menubarBackground and use that
`backgroundColor: styleVariables.menubarBackground`
And add that colour to the global variables,
`export const styleVariables = {
  backgroundColour: "#f5f5f5",
  unselected: "lightgrey",
  singleSelected: "lightblue",
  doubleSelected: "darkturquoise",
  menubarBackground: "#A9A9A9"
};`

pablisch commented on frontend/package.json:
```diff
@@ -9,11 +9,9 @@
   },
   "dependencies": {
     "@expo/vector-icons": "^14.0.0",
-    "@react-navigation/native": "^6.1.10",
```
This was in main previous to your branch and presumably is used somewhere. Easiest thing to do is run `npm install @react-navigation/native` to reinstall it.

pablisch commented on frontend/src/components/Button/ButtonStyles.js:
```diff
@@ -3,14 +3,14 @@ import { StyleSheet } from "react-native";
 export const styles = StyleSheet.create({
   container: {
     elevation: 8,
-    backgroundColor: "#2E8B57",
```
I think it is probably best to revert all changes to the button component for now. It was not intended as a part of this ticket to add any buttons as they will need to be displayed by context. Buttons will always need to be different sizes, colours, etc. so we will have to look at how to pass these in as props to this component.

pablisch commented on frontend/src/components/MenuBar/MenuBar.js:
```diff
@@ -0,0 +1,31 @@
+  const navigation = useNavigation();
+  return (
+    <View style={styles.container}>
+      <Button title="Clear all" />
```
As mentioned, this was not a part of this ticket but great to have them here, just don't worry about the styling.

pablisch commented on frontend/src/components/MenuBar/MenuBar.js:
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import { StyleSheet } from "react-native";
+//import { TouchableOpacity, Text } from "react-native";
```
There are a lot of comments and unused code, e.g. TouchableHighlight and Text are imported but not used. In general, unless there is some reason, I would get rid of unused code as when left, nobody knows why it is there and it tends to stay forever.

pablisch reviewed (changes requested):
It was a confusingly small ticket. 
Worth implementing the colour variable now I think although thinking about, probably best to stick with Ju's colour unless you have some reason for changing it. I'm not sure about the green on the plan personally but we should assume that the main screen will have some colour, or maybe not? Might be worth a discussion?

Please replace the missing dependency and delete any code you don't want. If you are sure it is needed then you can leave a comment about it.

If any comments are confusing, please ask.

pablisch commented on package.json:
```diff
@@ -16,5 +16,8 @@
   "license": "ISC",
   "devDependencies": {
     "husky": "^9.0.11"
+  },
```
This was installed in the root folder of the project, so should be removed from here where there should be no dependencies, and be installed in the frontend instead.

pablisch commented on frontend/src/components/MenuBar/MenuBar.js:
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import { StyleSheet } from "react-native";
+import { useNavigation } from "@react-navigation/native";
+import { View, Text, TouchableHighlight } from "react-native";
```
Do we need Text and TouchableHighlight here? I don't imagine we will use Text here and I have no clue what TouchableHighlight is.

Navigation is also imported and declared but never used. If you think we will use it later that is fine.

pablisch reviewed (changes requested):
Almost there!

kikilondon commented:
@pablisch I removed react-navigation/native dependency  from the package.json root folder and moved to frontend folder. 

I don't know anything about husky dependency, maybe there is a specific reason why they installed it there.If not, please let me know, and I will move it to the package.json front-end folder.

pablisch commented:
> Member

Don't worry about husky. I don't know what it is but it is something Sidra is doing about pre-commit hooks which is all a mystery to me. I imagine it relates to FE and BE so is in the root. I should have been clearer about meaning dependencies as opposed to dev-dependencies. 

I see that @react-navigation/native is back in the FE but I also still see it in the root. Since there should be no dependencies there all of the selected code here should go.

![Screenshot 2024-03-13 at 12 28 50](https://github.com/Pensano-dev/aura-mobile-app/assets/19639889/5cf84948-06c6-4d3d-b2f7-937eb24d2ce8)

If that is not showing on your local branch, check you have pushed everything (I keep forgetting this!) and if it's not that, let me know and we'll sort it.

Everything else looks ready to go.

pablisch reviewed (approved):
Great!

kikilondon commented:
Great ! Finally 🤣🤣




On Fri, 15 Mar 2024 at 19:33, Pablo Joyce ***@***.***> wrote:

> ***@***.**** approved this pull request.
>
> Great!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Pensano-dev/aura-mobile-app/pull/81#pullrequestreview-1939994964>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZZFRT3JPJBQGXKVEN7O6BLYYM5F7AVCNFSM6AAAAABESZ7IQSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMZZHE4TIOJWGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
